### PR TITLE
LG-2517 TextInput search type `aria-label`

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Fixes a regression when `type="search"`. Providing `aria-label` is sufficient for accessibility.

Updates TS types. Any one of `label`, `aria-label` and `aria-labelledby` is sufficient for all text inputs

🎟 _Jira ticket:_ [LG-2517](https://jira.mongodb.org/browse/LG-2517)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/text-input/src/TextInput.spec.tsx [hunk 1]
```diff
@@ -256,11 +256,18 @@ describe('packages/text-input', () => {
     test('TextInput throws error when neither aria-labelledby or label is supplied', () => {
       // @ts-expect-error
       <TextInput />;
+      // @ts-expect-error
+      <TextInput aria-label="some label" />;
```
Comment (TheSonOfThomp) [inline packages/text-input/src/TextInput.spec.tsx:260]: Not sure about this case @DesignerDave

File: packages/text-input/src/TextInput.tsx [hunk 2]
```diff
@@ -57,8 +57,12 @@ import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
  * @param props.sizeVariant determines the size of the text and the height of the input.
  */
 
-type TextInput = React.ForwardRefExoticComponent<TextInputProps>;
-const TextInput: TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+type TextInputComponentType = React.ForwardRefExoticComponent<TextInputProps>;
+/// @ts-expect-error
```
Comment (TheSonOfThomp) [inline packages/text-input/src/TextInput.tsx:61]: This is a temporary fix until the `SearchInput` is released

File: packages/text-input/src/TextInput.spec.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/text-input/src/TextInput.spec.tsx:260]: This should be allowed AFAICT

File: .changeset/spicy-donuts-talk.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/text-input': patch
+---
+
+Fixes a regression. Now when `type="search"`, providing `aria-label` is sufficient for accessibility
```
Comment (brookescarlett) [inline .changeset/spicy-donuts-talk.md:5]: Aria-label should always be sufficient for accessibility as it relates to TextInput components

File: packages/text-input/src/TextInput.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import React, { forwardRef, useState } from 'react';
+import React, { useState } from 'react';
```
Comment (brookescarlett) [inline packages/text-input/src/TextInput.tsx:1]: Is removing this related to the typescript changes?

File: packages/text-input/src/types.ts [hunk 5]
```diff
@@ -38,26 +38,41 @@ export const TextInputFontSize = {
...
+  ['aria-labelledby']?: string;
+}
+
+type AriaLabels = keyof AriaLabelProps;
+
+interface TextInputTypeProp {
+  /**
+   * The input type.
+   *
+   * Requires an `aria-label` if `"search"` is provided
```
Comment (brookescarlett) [inline packages/text-input/src/types.ts:63]: Aria-labelledby can be sufficient here as well. We don't necessarily need aria-label if search is provided.

File: packages/text-input/src/TextInput.tsx (hunk 4 above)
Comment (TheSonOfThomp) [reply packages/text-input/src/TextInput.tsx:1]: Just removing the explicit import in favor of `React.forwardRef`

File: packages/text-input/src/TextInput.spec.tsx (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/text-input/src/TextInput.spec.tsx]: 👍 Updated to require one of `label`, `aria-label`, or `aria-labelledby` in all cases

**Review by brookescarlett — APPROVED** [review]

**Pull request merged**